processor/transformprocessor/internal/metrics/func_aggregate_on_attributes_metrics.go

```diff
@@ -0,0 +1,59 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package metrics // import "github.com/open-telemetry/opentelemetry-collector-contrib/processor/transformprocessor/internal/metrics"
+
+import (
+	"context"
+	"fmt"
+
+	"go.opentelemetry.io/collector/pdata/pmetric"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/coreinternal/aggregateutil"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/contexts/ottlmetric"
+)
+
+type aggregateOnAttributesArguments struct {
+	Type       string
+	Attributes ottl.Optional[[]string]
+}
+
+func newAggregateOnAttributesFactory() ottl.Factory[ottlmetric.TransformContext] {
+	return ottl.NewFactory("aggregate_on_attributes", &aggregateOnAttributesArguments{}, createAggregateOnAttributesFunction)
+}
+
+func createAggregateOnAttributesFunction(_ ottl.FunctionContext, oArgs ottl.Arguments) (ottl.ExprFunc[ottlmetric.TransformContext], error) {
+	args, ok := oArgs.(*aggregateOnAttributesArguments)
+
+	if !ok {
+		return nil, fmt.Errorf("AggregateOnAttributesFactory args must be of type *AggregateOnAttributesArguments")
+	}
+
+	t, err := aggregateutil.ConvertToAggregationType(args.Type)
+	if err != nil {
+		return nil, fmt.Errorf("invalid aggregation function: %s", err.Error())
```

Comment:
Lets include in the error the valid values.  Lets also specify in the README that `function` is case-sensitive.

Reply:
Alternatively, could we make it case-insensitive and convert it to lowercase here?

Reply:
we don't do other stuff case-insensitive (such as merge mode in merge_maps).

Reply:
adapted!

Reply:
I would be okay making everything case-insensitive; it's easy to implement and I think would improve the user experience.

That said, we can add that as an enhancement later if we care to.